packages/typespec-vscode/src/emit/emitter.ts

```diff
@@ -0,0 +1,42 @@
+import vscode from "vscode";
+import { SettingName } from "../types.js";
+
+export enum EmitterKind {
+  Schema = "schema",
+  Client = "client",
+  Server = "server",
+}
+
+export interface Emitter {
+  language: string;
+  package: string;
+  version?: string;
+  kind: EmitterKind;
+}
+
+const extensionConfig = vscode.workspace.getConfiguration();
```

Comment:
move to getRegisteredEmitters? user may change the configure any time, so you need to get it everytime when user trigger the feature.